fluentgen/src/test/java/com/azure/autorest/fluent/DeduplicateSetterTest.java

```diff
@@ -0,0 +1,123 @@
+import org.yaml.snakeyaml.nodes.Tag;
+import org.yaml.snakeyaml.representer.Representer;
+
+import java.util.List;
+
+public class DeduplicateSetterTest {
+
+    private final JavaSettings settings = JavaSettings.getInstance();
+    private final JavaFileFactory javaFileFactory = new JavaFileFactory(settings);
+
+    @BeforeAll
+    public static void init(){
+        TestUtils.MockFluentGen fluentgen = new TestUtils.MockFluentGen();
+        fluentgen.setValue("model-override-setter-from-superclass", true);
+    }
+
+    /**
+     * Issue: https://github.com/Azure/autorest.java/issues/1320
+     * Remove duplicate setter methods from child schema when parent schema contains same property
+     * mainly to test {@link com.azure.autorest.template.ModelTemplate#getParentSettersToOverride(ClientModel, JavaSettings, List)}
+     */
+    @Test
+    public void deduplicateTest(){
+        //construct parent schema
+        ObjectSchema siteSchemaParent = loadSiteSchemaParent();
+        ClientModel modelParent = ModelMapper.getInstance().map(siteSchemaParent);
+        //get child schema
+        ClientModel model = modelParent.getDerivedModels().get(0);
+        ModelTemplateAccessor templateAccessor = new ModelTemplateAccessor();
+        List<ClientModelPropertyReference> propertyReferences = templateAccessor.getClientModelPropertyReferences0(modelParent.getDerivedModels().get(0));
+        if (!CoreUtils.isNullOrEmpty(model.getPropertyReferences())) {
+            propertyReferences.addAll(model.getPropertyReferences());
+        }
+        // reference to properties from parent model
+        JavaFile javaFile = javaFileFactory.createSourceFile("com.azure.site", "Test");
+        String classNameWithBaseType = model.getName();
+        if (model.getParentModelName() != null) {
+            classNameWithBaseType += String.format(" extends %1$s", model.getParentModelName());
+        }
+        javaFile.publicClass(Lists.newArrayList(), classNameWithBaseType, classBlock -> {
+            // real test here
+            List<ClientModelPropertyAccess> toOverride = templateAccessor.getParentSettersToOverride(model, settings, propertyReferences);
+            Assertions.assertEquals(toOverride.size(), 1);
+
+            for (ClientModelPropertyAccess parentProperty : toOverride) {
+                classBlock.javadocComment(JavaJavadocComment::inheritDoc);
+                classBlock.annotation("Override");
+                classBlock.publicMethod(String.format("%s %s(%s %s)",
+                        model.getName(),
+                        parentProperty.getSetterName(),
+                        parentProperty.getClientType(),
+                        parentProperty.getName()),
+                    methodBlock -> {
+                        methodBlock.line(String.format("super.%1$s(%2$s);", parentProperty.getSetterName(), parentProperty.getName()));
+                        methodBlock.methodReturn("this");
+                    });
+            }
```

Comment:
I didn't get why this is a test. It just write code?

For these kind of issue from real swagger, I usually just put it to a new part in `prepare-tests.bat`
https://github.com/Azure/autorest.java/blob/main/fluent-tests/prepare-tests.bat#L34-L36
fluent-tests will make sure code generated build pass.

Reply:
Deleted.